packages/next/src/server/lib/incremental-cache/index.ts

```diff
@@ -589,4 +596,33 @@ export class IncrementalCache implements IncrementalCacheType {
       console.warn('Failed to update prerender cache for', pathname, error)
     }
   }
+
+  checkTags({ softTags, tags }: GetIncrementalFetchCacheContext) {
+    const revalidatedTags = this.revalidatedTags
+    const workStore = workAsyncStorage.getStore()
+    const pendingRevalidatedTags = workStore?.pendingRevalidatedTags
+
+    let checker: ((tag: string) => boolean) | null = null
+    if (
+      revalidatedTags &&
+      revalidatedTags.length > 0 &&
+      pendingRevalidatedTags &&
+      pendingRevalidatedTags.length > 0
+    ) {
+      checker = (tag: string) =>
+        revalidatedTags.includes(tag) || pendingRevalidatedTags.includes(tag)
+    } else if (revalidatedTags && revalidatedTags.length > 0) {
+      checker = (tag: string) => revalidatedTags.includes(tag)
+    } else if (pendingRevalidatedTags && pendingRevalidatedTags.length > 0) {
```

Comment:
It's not immediately obvious that there can never be both `revalidatedTags` and `pendingRevalidatedTags` at the same time.